addons/web/static/tests/webclient/user_menu_tests.js

```diff
@@ -129,3 +132,42 @@ QUnit.test("display the correct name in debug mode", async (assert) => {
   assert.containsOnce(userMenuEl, "span.oe_topbar_name");
   assert.strictEqual(userMenuEl.querySelector(".oe_topbar_name").textContent, "Sauron (test)");
 });
+
+QUnit.test("can execute the callback of settings", async (assert) => {
+  const mockRPC = (route) => {
+    if (route === "/web/dataset/call_kw/res.users/action_get") {
+      return Promise.resolve({
+        name: "Change My Preferences",
+        res_id: 0,
+      });
+    }
+  };
+  const testConfig = { mockRPC };
+  serviceRegistry.add("localization", makeFakeLocalizationService());
+  serviceRegistry.add("orm", ormService);
+  const fakeActionService = {
+    name: "action",
+    start() {
+      return {
+        doAction(actionId) {
+          assert.step("" + actionId.res_id);
+          assert.step(actionId.name);
+          return Promise.resolve(true);
+        },
+      };
+    },
+  };
+  serviceRegistry.add("action", fakeActionService, { force: true });
+
+  env = await makeTestEnv(testConfig);
+  userMenuRegistry.add("profile", preferencesItem);
+  userMenu = await mount(UserMenu, { env, target });
+  let userMenuEl = userMenu.el;
```

Comment:
userMenuEl not really necessary, you can remove l 165 and l 167 and use userMenu.el everywhere